Bin/Stub.php

```diff
@@ -0,0 +1,208 @@
+ */
+
+class Stub extends \Hoa\Console\Dispatcher\Kit {
+
+    /**
+     * Options description.
+     *
+     * @var \Hoa\Devtools\Bin\Stub array
+     */
+    protected $options = [
+        ['dry-run',      Console\GetOption::NO_ARGUMENT      , 'd'],
+        ['verbose',      Console\GetOption::NO_ARGUMENT      , 'V'],
+        ['stub',         Console\GetOption::REQUIRED_ARGUMENT, 's'],
+        ['help',         Console\GetOption::NO_ARGUMENT      , 'h'],
+        ['help',         Console\GetOption::NO_ARGUMENT      , '?']
+    ];
+
+
+
+    /**
+     * The entry method.
+     *
+     * @access  public
+     * @return  int
+     */
+    public function main ( ) {
+
+        $porcelaine = false;
+        $stub       = null;
+        $verbose    = false;
+
+        while(false !== $c = $this->getOption($v)) switch($c) {
+
+            case 'V':
+                $verbose = true;
+                break;
+
+            case 'd':
+                $porcelaine = true;
+              break;
+            
+            case 's':
+                $stub = $v;
+                break;
+
+            case 'h':
+            case '?':
+                return $this->usage();
+              break;
+
+            case '__ambiguous':
+                $this->resolveOptionAmbiguity($v);
+              break;
+        }
+
+        $hoaPath = 'hoa://Library/';
+
+        if($stub === null)
+            return $this->usage();
+
+        $finder = new \Hoa\File\Finder();
+        $finder
+            ->in($hoaPath)
+            ->name('#\.php$#')
+            ->maxDepth(100)
+            ->files();
+
+        $f = array();
+
+        // READ
+        foreach ($finder as $key => $value) {
+            $pathname = $value->getPathName();
+            $name     = $value->getBasename();
+
+            preg_match('#Library([^\.]*)\.php$#', $pathname, $m);
+
+            if(isset($m[1]) === true)
+            {
+                $name = 'Hoa'.$m[1];
+
+                $e = [];
+                foreach (explode('/', $name) as $value) {
+                    $e[] = ucfirst($value);
+                }
+
+
+                $name = implode('/', $e);
+                $end  = '';
+
+                $raw = file_get_contents($pathname);
+
+                 preg_match('#\nclass_alias\(([^,]*),(.*)\)#', $raw, $classAlias);
+
+                if (isset($classAlias[1]) and $classAlias[1] !== '') {
+                    $name  = substr(trim($classAlias[1]), 1, -1);
+                    $end   = substr(trim($classAlias[2]), 1, -1);
+                }
+                 
+                preg_match('#flexEntity\(\'(.*)\'#', $raw, $flexEntity);
+
+                if (count($flexEntity) > 0) {
+                    $c   = $flexEntity[1];
+                    $end = \Hoa\Core\Consistency\Consistency::getEntityShortestName($c);
+                }
+
+                if ($end !== '') {
+                    $f[] = array('class' => str_replace('/', '\\', $name), 'alias' => $end);
```

Comment:
In Hoa, there may be more than one space between `class` keyword and class name.
eg: [`Hoa\Response`](https://github.com/hoaproject/Http/blob/master/Response/Response.php#L82).
```php
'#((?:abstract\s)?class|interface|trait)\s+'.$c.'\s#'
```

(However it's not [PSR-2](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md#41-extends-and-implements).)